src/urllib3/util/url.py

```diff
@@ -155,6 +158,11 @@ def parse_url(url):
         # Empty
         return Url()
 
+    # Prevent CVE-2019-9740.
+    # adapted from https://github.com/python/cpython/pull/12755
+    if _contains_disallowed_url_pchar_re.search(url):
+        raise LocationParseError("URL can't contain control characters. {}".format(url))
```

Comment:
I recommend using a repr of the URL rather than the raw url containing the offending text in the error message.  `{!r}` iirc.